Switch to upstream ghaction-import-gpg action

Reference: https://github.com/hashicorp/ghaction-import-gpg/issues/11

The GitHub hosted runners are now running gpg-agent by default, which can cause the following error:

```
Run hashicorp/ghaction-import-gpg@v2.1.0
Run gpg-agent --daemon --default-cache-ttl 7200
gpg-agent[1632]: directory '/home/runner/.gnupg' created
gpg-agent[1632]: directory '/home/runner/.gnupg/private-keys-v1.d' created
gpg-agent: a gpg-agent is already running - not starting a new one
Error: Process completed with exit code 2.
```

The `openscientia` fork of the `hashicorp/ghaction-import-gpg` action that bypassed this error is no longer needed as the [upstream](https://github.com/crazy-max/ghaction-import-gpg) of the original fork (`hashicorp/ghaction-import-gpg`) already resolves the issue and more.

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -51,10 +51,10 @@ jobs:
           go-version: ${{ needs.go-version.outputs.version }}
       - name: Import GPG Key
         id: import_gpg
-        uses: openscientia/ghaction-import-gpg@v2.1.2
-        env:
-          GPG_PRIVATE_KEY: ${{ secrets.GPG_PRIVATE_KEY }}
-          PASSPHRASE: ${{ secrets.PASSPHRASE }}
+        uses: crazy-max/ghaction-import-gpg@v5
+        with:
+          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+          passphrase: ${{ secrets.PASSPHRASE }}
       - name: Release Notes Download
         id: release-notes-download
         uses: actions/download-artifact@v3

